### PR TITLE
added simple SDU Portal Authentication backend

### PR DESCRIPTION
**Pull request by YernarKambar**

### CODE REVIEW

File: sdu_beta_career/auth_backends.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+from django.contrib.auth.models import User
+
+
+class SDUPortalAuthenticationBackend:
+
+    def authenticate(self, username=None, password=None):
+        """
+        Try to find user in SDU portal database and if that user exists, then to create in SBC database
+        :param username: Username or maybe email of user
+        :type username: str
```
Comment (An4ik) [inline sdu_beta_career/auth_backends.py]: we have typing module now, no need to write it in doc.

so it should be like this:

`def authenticate(self, username:str=None, password:str =None) -> Optional[User]`

File: sdu_beta_career/auth_backends.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+        :param password: password of user
+        :type password: str or maybe encrypted str
+
+        :return: user from SDU portal database
+        :rtype: instance of a class
+        """
+        try:
+            if username == username and password == password:
+                user = User.objects.create_user(username=username, password=password)
```
Comment (An4ik) [inline sdu_beta_career/auth_backends.py]: maybe
`User.objects.create(username=username, password=password, first_name="Bla-Bla, last_name="Bla-Bla")  is better ?`

Comment (YernarKambar) [reply sdu_beta_career/auth_backends.py]: you are right! thanks)

**Comment by YernarKambar** [comment]
check my changes plz

File: sdu_beta_career/auth_backends.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            # difference between an existing and a nonexistent user (#20760).
+            UserModel().set_password(password)
+        else:
+            if not(user.check_password(password) and self.user_can_authenticate(user)):
+                # Try to find user in SDU portal database
+                user = self.get_portal_user(username, password)
+            return user
+
+    @staticmethod
+    def get_portal_user(username: str=None, password: str=None) -> 'Optional[User]':
```
Comment (An4ik) [inline sdu_beta_career/auth_backends.py]: 'Optional[User]'   why there are quotes ?

Comment (An4ik) [inline sdu_beta_career/auth_backends.py]: username: Optional[str], password: Optional[str]  is the right

File: sdu_beta_career/auth_backends.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        if username == username and password == password:
+            user = User.objects.create_user(username=username,
+                                            password=password,
+                                            first_name='TestFirstName',
+                                            last_name='TestLastName')
+            user.save()
```
Comment (An4ik) [inline sdu_beta_career/auth_backends.py]: no need to save since created

File: sdu_beta_career/auth_backends.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            return user
+        else:
```
Comment (An4ik) [inline sdu_beta_career/auth_backends.py]: in python function returns None by default , so you can remove else statement

**Pull request merged**